### DISCUSSION TITLE
i need your help please in this hard idea

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5579] [Help]
hi everyone how are you ?

i have hard idea want to execute and need your help in it please 

i previously add discussion about 2 forms in 1 but i was so wrong not this idea is required

the required idea is i have 2 row (groups, voting ) when i click on groups the grid of group will open normally, press on new group dailouge will open to insert name of groups 
enter the name and press save all that normally 
 
![Screenshot (118)](https://user-images.githubusercontent.com/77009503/110619406-74d81200-81a0-11eb-8cbb-570429c6da28.png)

![Screenshot (117)](https://user-images.githubusercontent.com/77009503/110619459-83bec480-81a0-11eb-984f-ceb05dde33d6.png)

now that what i need 

when i press on group1 it supposed open the following page 

![Screenshot (119)](https://user-images.githubusercontent.com/77009503/110619688-c2ed1580-81a0-11eb-9e2d-a75fa7b618ad.png)

but retrived only the votes that related to that group so if there is no votes related to that group it open empty and i can create new one 

how can i execute that hard idea :(

**Comment by choss21** [comment]
Hello

look at the examples:
[Part1](https://serenity.is/demo/BasicSamples/CustomLinksInGrid) 

in part2 you can passs query params 
[Part2](https://serenity.is/demo/Northwind/Product?cat=1) Check typescript file

**Comment by luongnv6** [comment]
in GroupGrid
```C#
protected getColumns() {
            var columns = super.getColumns();

            Q.first(columns, x => x.field == 'GroupName').format = ctx => {
                let item: GroupRow = ctx.item;
                return `<a href="/Dashboard/Voting?groupId=${item.Id}" class="view-edit-group">${item.GroupName}</a>`;
            }

            return columns;
        }
```

in VotingGrid
```C#
 protected onViewSubmit() {
            if (!super.onViewSubmit()) {
                return false;
            }
            var q = Q.parseQueryString();
            var request = this.view.params as Serenity.ListRequest;
            if (q["groupId"]) {
                  request.Criteria = Serenity.Criteria.and(request.Criteria, [['GroupId'], '=', q["groupId"]]);
            } else  {
                  request.Criteria = Serenity.Criteria.and(request.Criteria, [['GroupId'], '=', -999]);
            }
            return true;
        }
```

**Reply by samaremad** [reply]
i get error when i add code to group grid

![Screenshot (121)](https://user-images.githubusercontent.com/77009503/111150261-8a7d7b00-8596-11eb-9f92-ee25edb0421b.png)


![Screenshot (122)](https://user-images.githubusercontent.com/77009503/111150274-8fdac580-8596-11eb-9636-8589295d5510.png)


![Screenshot (123)](https://user-images.githubusercontent.com/77009503/111150290-936e4c80-8596-11eb-862f-87935a52f53e.png)
